aiida/orm/nodes/data/array/bands.py

```diff
@@ -885,7 +891,10 @@ def _prepare_gnuplot(self,
         """
         import os
 
-        dat_filename = os.path.splitext(main_file_name)[0] + '_data.dat'
+        if main_file_name is not None:
```

Comment:
By the way, the default seems to be `''` rather than `None`

I suggest to change the default of the function argument to `None`, and then:
```python
if not main_file_name:  # also catches '', just to be safe
    main_file_name = 'band.dat'  # not sure what a reasonable default is
dat_filename = os.path.splitext(main_file_name)[0] + '_data.dat'
```

Reply:
the default filename is not important here, in fact it is not used. Default filename is only used when export gnuplot without specify the extra_file with `-o`, if that happened it will throw a command line critical information. Here the file_name is only to make sure `os.path.splitext` is working and throw the right Critical information to user.